src/components/ScenarioParameters/UploadFileUtils.js

```diff
@@ -3,13 +3,38 @@
 
 import React from 'react';
 import { UPLOAD_FILE_STATUS_KEY } from '@cosmotech/ui';
-import { ORGANIZATION_ID, WORKSPACE_ID } from '../../config/AppInstance';
+import {
+  ORGANIZATION_ID, WORKSPACE_ID,
+  ENABLE_APPLICATION_INSIGHTS
+} from '../../config/AppInstance';
 import DatasetService from '../../services/dataset/DatasetService';
 import WorkspaceService from '../../services/workspace/WorkspaceService';
 import { FileUpload } from './components/tabs';
+import { AppInsights } from '../../services/AppInsights';
 
 export const STORAGE_ROOT_DIR_PLACEHOLDER = '%WORKSPACE_FILE%/';
 
+// App Insigths
+const appInsights =
+  ENABLE_APPLICATION_INSIGHTS
+    ? AppInsights.getInstance()
+    : undefined;
+
+// Track upload
+const trackUpload = (scenarioId) => {
+  if (appInsights) {
+    appInsights.trackEvent({ name: 'UploadFile' }, { name: scenarioId });
+    appInsights.trackMetric({ name: 'UploadFileValue', average: 1, sampleCount: 1 });
+  }
+};
+  // Track download
+const trackDownload = (scenarioId) => {
+  if (appInsights) {
+    appInsights.trackEvent({ name: 'DownloadFile' }, { scenarioId: scenarioId });
+    appInsights.trackMetric({ name: 'DownloadFileValue', average: 1, sampleCount: 1 });
+  }
```

Comment:
We should hide the implementation details of the calls to `trackEvent` and `trackMetric` in the service instead of using it in utils files and components.

Can we move `trackUpload` and `trackDownload` methods in the `AppInsights` service ?